Sources/OktaAnalytics/AnalyticsProviderProtocol.swift

```diff
@@ -11,37 +11,49 @@
  */
 
 import Foundation
+import Combine
 import OktaLogger
 
 /**
  Protocol: Client needs to implement this protocol and provide the information about provider
  */
-@objc
 public protocol AnalyticsProviderProtocol: AnyObject {
 
-    /// provider name, eg: Firebase, AppCenter etc.
+    /// The name of the provider, e.g. Firebase, AppCenter, etc.
     var name: String { get set }
-    /// logger to post logs on it's destination
+
+    /// The logger to post logs on its destination
     var logger: OktaLoggerProtocol? { get set }
-    /// default properties that post with all events
+
+    /// The default properties that are posted with all events
     var defaultProperties: [String: String]? { get set }
-    /// Tracks event to the provider
-    func trackEvent(_ eventName: String, withProperties: [String: String]?)
+
+    /// A dictionary of event scenarios
+    var scenarios: [EventName: EventScenario]? { get set }
+
+    /// Tracks an event to the provider
+    func trackEvent(_ eventName: EventName, withProperties: [String: String]?)
+
+    /// Starts an event scenario
+    func startScenario(_ eventName: EventName, _ propertySubject: (PassthroughSubject<Property, Never>?) -> Void)
+
+    /// Adds an event to an event scenario
+    func addEventToScenario(_ eventName: EventName, _ propertySubject: (PassthroughSubject<Property, Never>?) -> Void)
```

Comment:
Do we throw exception if the scenario is not found or started?

Reply:
Actually, we don't need that because `propertySubject ` is optional and we will write in code comments and documentation about that